src/app/pages/student-display/student-display.component.html

```diff
@@ -1,35 +1,35 @@
 <ng-container [ngSwitch]="currentScreen$ | async">
-  <ng-container *ngSwitchCase="ScreenId.SessionLobby">
+
+  <app-student-round *ngSwitchCase="ScreenId.StudentRound"></app-student-round>
+
+  <ng-container *ngSwitchDefault>
     <app-hill-background cover></app-hill-background>
-    <div class="centered-container">
-      <mat-card>
+    <div class="centered-container" [ngSwitch]="currentScreen$ | async">
+
+      <mat-card *ngSwitchCase="ScreenId.SessionLobby">
         <mat-card-title>Waiting for round to begin&hellip;</mat-card-title>
+        <mat-card-content>
+          Please wait for the host to start a round.
+        </mat-card-content>
+        <mat-card-actions align="end">
+          <button mat-button routerLink="/">Exit Session</button>
+        </mat-card-actions>
       </mat-card>
-    </div>
-  </ng-container>
 
-  <ng-container *ngSwitchCase="ScreenId.StudentRound">
-    <app-student-round></app-student-round>
-  </ng-container>
-
-  <ng-container *ngSwitchCase="ScreenId.InvalidSession">
-    <app-hill-background cover></app-hill-background>
-    <div class="centered-container">
-      <mat-card>
+      <mat-card *ngSwitchCase="ScreenId.InvalidSession">
         <mat-card-title>Error: invalid session</mat-card-title>
+        <mat-card-actions align="end">
+          <button mat-button color="primary" routerLink="/">Go Back</button>
+        </mat-card-actions>
       </mat-card>
-    </div>
-	</ng-container>
 
-  <ng-container *ngSwitchDefault>
-    <app-hill-background cover></app-hill-background>
-    <div class="centered-container">
-      <mat-card>
+      <mat-card *ngSwitchDefault>
         <mat-card-title>
           Loading Session&hellip;
         </mat-card-title>
         <mat-spinner></mat-spinner>
       </mat-card>
+
     </div>
   </ng-container>
 
```

Comment:
Ah, okay, I see what you did here.

I'd prefer to avoid nested `ngSwitch`s if it's possible, though—HTML tends to be very deeply indented, and it can be difficult to identify whether a given case belongs to the inner or outer switch. I know we would have to write the `<app-hill-background cover></app-hill-background>` element once for each NgSwitchCase, but I think the extra readability would be worth it.

If you disagree, though, I'd also be okay with leaving it as it is :)

Reply:
I was noticing that as a student since these screens tend to be switched out quite a bit it really didn't look good and was distracting when it flickered white everytime something changed. Even with this change there is still a bit of flicker that can happen when transitioning between session lobby and round. Honestly we may want to refactor the entire student display to not use the nested `student-round` component.

Reply:
Oh, okay, if there's a practical benefit, then I'm all for doing things this way. Thanks for explaining that! 🙂